File: src/data/employee.ts
import { Employee, Department } from '../types/employee';

export const departments: Department[] = [
  {
    id: 'marketing',
    name: 'Marketing',
    employeeCount: 12,
    color: 'bg-amber-100 text-amber-800',
    icon: 'briefcase'
  },
  {
    id: 'sales',
    name: 'Sales',
    employeeCount: 18,
    color: 'bg-blue-100 text-blue-800',
    icon: 'briefcase'
  },
  {
    id: 'engineering',
    name: 'Engineering',
    employeeCount: 24,
    color: 'bg-green-100 text-green-800',
    icon: 'briefcase'
  }
];

export const employees: Employee[] = [
  {
    id: '1',
    name: 'John Smith',
    position: 'Marketing Manager',
    department: 'Marketing',
    email: 'john@gmail.com',
    phone: '+91 1236574536',
    joinedDate: 'May 18 2025'
  },
  {
    id: '2',
    name: 'Sara Joseph',
    position: 'Sales Manager',
    department: 'Sales',
    email: 'sara@gmail.com',
    phone: '+91 1236574536',
    joinedDate: 'May 18 2024'
  },
  {
    id: '3',
    name: 'Michael Chen',
    position: 'Lead Developer',
    department: 'Engineering',
    email: 'michael@gmail.com',
    phone: '+91 1236574536',
    joinedDate: 'Jan 15 2024'
  },
  {
    id: '4',
    name: 'Emma Wilson',
    position: 'Marketing Specialist',
    department: 'Marketing',
    email: 'emma@gmail.com',
    phone: '+91 1236574536',
    joinedDate: 'Mar 10 2024'
  },
  {
    id: '5',
    name: 'David Brown',
    position: 'Sales Executive',
    department: 'Sales',
    email: 'david@gmail.com',
    phone: '+91 1236574536',
    joinedDate: 'Feb 20 2024'
  },
  {
    id: '6',
    name: 'Lisa Anderson',
    position: 'Software Engineer',
    department: 'Engineering',
    email: 'lisa@gmail.com',
    phone: '+91 1236574536',
    joinedDate: 'Apr 05 2024'
  }
];

File: lib/models/employee.dart
class Employee {
  final String id;
  final String name;
  final String empId;
  final String department;
  final String email;
  final String phone;
  final String? position;
  final String? joinedDate;

  Employee({
    required this.id,
    required this.name,
    required this.empId,
    required this.department,
    required this.email,
    required this.phone,
    this.position,
    this.joinedDate,
  });

  factory Employee.fromJson(Map<String, dynamic> json) {
    return Employee(
      id: json['id'],
      name: json['name'],
      empId: json['empId'],
      department: json['department'],
      email: json['email'],
      phone: json['phone'],
      position: json['position'],
      joinedDate: json['joinedDate'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'empId': empId,
      'department': department,
      'email': email,
      'phone': phone,
      'position': position,
      'joinedDate': joinedDate,
    };
  }
}

class Department {
  final String id;
  final String name;
  final int employeeCount;
  final String color;
  final String icon;

  Department({
    required this.id,
    required this.name,
    required this.employeeCount,
    required this.color,
    required this.icon,
  });
}